File: saas-admin/src/types/planPrice.ts
import { Status, CurrencyCode } from "./app";

export type PlanPriceDto = {
  id: string;
  status: Status;
  createdDate: string;
  modifiedDate: string;
  autoID: number;
  planId: string;
  currency: CurrencyCode;
  amount: number;
  effectiveFrom: string;
  effectiveTo: string | null;
  isCurrent: boolean;
};

export type PlanPriceAddDto = {
  planId: string;
  currency: CurrencyCode;
  amount: number;
  effectiveFrom?: string;
  effectiveTo?: string | null;
  isCurrent?: boolean;
};

export type PlanPriceUpdateDto = {
  id: string;
  planId: string;
  currency: CurrencyCode;
  amount: number;
  effectiveFrom: string;
  effectiveTo?: string | null;
  isCurrent: boolean;
};

File: SaasV2.Entity/PlanPrice.cs
using SaasV2.Core.Enums;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace SaasV2.Entity
{
    // Plan fiyatı (geçmişe dönük fiyat izlemi için tarih aralıklı)
    public class PlanPrice : BaseEntity
    {
        public Guid PlanId { get; set; }
        public CurrencyCode Currency { get; set; } = CurrencyCode.TRY;
        public decimal Amount { get; set; }            // Vergi hariç/ dahil, stratejine göre

        public DateTime EffectiveFrom { get; set; }    // yürürlük başlangıcı
        public DateTime? EffectiveTo { get; set; }     // null = halen geçerli
        public bool IsCurrent { get; set; }            // kolay filtre için
    }
}
